src/main/ApplicationUtils.cpp

```diff
@@ -56,7 +57,26 @@ runWithConfig(Config cfg)
                          << "(for testing only)";
         }
 
-        app->start();
+        if (cc)
+        {
+            Json::Value catchupInfo;
+            std::shared_ptr<HistoryArchive> archive;
+            auto const& ham = app->getHistoryArchiveManager();
+            archive = ham.selectRandomReadableHistoryArchive();
+            if (catchup(app, *cc, catchupInfo, archive) == 0)
+            {
+                // Pre-run catchup defers overlay startup.
```

Comment:
nit: looks to me that this code should be moved to line 80 with something like `if (!cfg.MODE_AUTO_STARTS_OVERLAY) {...}` as to make it future proof

Reply:
Yes, fixed.